articles/remapping.md

```diff
@@ -0,0 +1,77 @@
+### Remapping Names
+#### ROS 1
+In ROS 1 [Names](http://wiki.ros.org/Names) can be remapped to other names by passing [remapping arguments](http://wiki.ros.org/Remapping%20Arguments) via the command line. `roscpp` additionally allows remaps to be passed via Init as strings in STL containers. A ROS1 remap consists of two names: one that should be replaced with another.
+
+In ROS1 remapping works with FQN. When a name is to be checked for remappings, it is first expanded to a FQN. Both parts of the remapping rule are expanded to a FQN. Then a name is changed to the remap's second name only if it exactly matches the first name.
+
+#### ROS 2
+In ROS2 remapping will be more powerful. There are more types of remappings that will be supported
+
+- Namespace replacement
+
+ - Replace all or part of a namespace with another namespace
+ 
+- Basename replacement
+
+ - Replace all of the basename with another basename
+ 
+- Arbitrary substring replacement
+
+ - Replace any substring with another string
+ 
+- Exact replacement (This is ROS 1)
+
+ - Replace a name with another name exactly
+
+ROS 2 may support remapping rules being applied either before or after the names are expanded to FQN. ROS 1 only supports the latter. 
+
+#### Pre-FQN expansion remapping
+Applying rules prior to FQN expansion enables this use case. Say the source code for a node use two names `"cat"` and `"/ns/cat"`. The node is in namespace `"ns"`. With just Post-FQN expantion it is impossible to remap these names individually. With Pre-FQN expansion remapping, a rule can be made to remap just one of them.
```

Comment:
> Adding on to the ROS 1 issue, I think you can do this in ROS 1 already.

Actually, you are probably right that ROS 1 cannot do this as-is, but it would be good to verify this.